pkg/validation/variables/variables_command_test.go

```diff
@@ -0,0 +1,149 @@
+				assert.Equal(t, expectedExecCommand, testExecCommand, "The two values should be the same.")
+			}
+		})
+	}
+}
+
+func TestValidateAndReplaceCompositeCommand(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		testFile     string
+		outputFile   string
+		variableFile string
+		wantErr      bool
+	}{
+		{
+			name:         "Good Substitution",
+			testFile:     "test-fixtures/commands/composite.yaml",
+			outputFile:   "test-fixtures/commands/composite-output.yaml",
+			variableFile: "test-fixtures/variables/variables-referenced.yaml",
+			wantErr:      false,
+		},
+		{
+			name:         "Invalid Reference",
+			testFile:     "test-fixtures/commands/composite.yaml",
+			outputFile:   "test-fixtures/commands/composite.yaml",
+			variableFile: "test-fixtures/variables/variables-notreferenced.yaml",
+			wantErr:      true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testCompositeCommand := v1alpha2.CompositeCommand{}
+			readFileToStruct(t, tt.testFile, &testCompositeCommand)
+
+			testVariable := make(map[string]string)
+			readFileToStruct(t, tt.variableFile, &testVariable)
+
+			err := validateAndReplaceForCompositeCommand(testVariable, &testCompositeCommand)
+			verr, ok := err.(*InvalidKeysError)
+			if tt.wantErr && !ok {
+				t.Errorf("Expected InvalidKeysError error from test but got %+v", verr)
+			} else if !tt.wantErr && err != nil {
+				t.Errorf("Got unexpected error: %s", err)
+			} else {
+				expectedCompositeCommand := v1alpha2.CompositeCommand{}
+				readFileToStruct(t, tt.outputFile, &expectedCompositeCommand)
+				assert.Equal(t, expectedCompositeCommand, testCompositeCommand, "The two values should be the same.")
+			}
+		})
+	}
+}
+
+func TestValidateAndReplaceApplyCommand(t *testing.T) {
+
+	tests := []struct {
+		name         string
+		testFile     string
+		outputFile   string
+		variableFile string
+		wantErr      bool
+	}{
+		{
+			name:         "Good Substitution",
+			testFile:     "test-fixtures/commands/apply.yaml",
+			outputFile:   "test-fixtures/commands/apply-output.yaml",
+			variableFile: "test-fixtures/variables/variables-referenced.yaml",
+			wantErr:      false,
+		},
+		{
+			name:         "Invalid Reference",
+			testFile:     "test-fixtures/commands/apply.yaml",
+			outputFile:   "test-fixtures/commands/apply.yaml",
+			variableFile: "test-fixtures/variables/variables-notreferenced.yaml",
+			wantErr:      true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testApplyCommand := v1alpha2.ApplyCommand{}
+			readFileToStruct(t, tt.testFile, &testApplyCommand)
+
+			testVariable := make(map[string]string)
+			readFileToStruct(t, tt.variableFile, &testVariable)
+
+			err := validateAndReplaceForApplyCommand(testVariable, &testApplyCommand)
+			verr, ok := err.(*InvalidKeysError)
+			if tt.wantErr && !ok {
+				t.Errorf("Expected InvalidKeysError error from test but got %+v", verr)
```

Comment:
If `!ok` here, you can't use the value of `verr` (should be `nil` I think)

Reply:
updated to `err` for all the tests